tensorflow/contrib/seq2seq/python/ops/attention_wrapper.py

```diff
@@ -1144,13 +1149,28 @@ def __init__(self,
             "If provided, attention_layer_size must contain exactly one "
             "integer per attention_mechanism, saw: %d vs %d"
             % (len(attention_layer_sizes), len(attention_mechanisms)))
+      if attention_layer_activation is not None:
+        attention_layer_activations = tuple(
+            attention_layer_activation
+            if isinstance(attention_layer_activation, (list, tuple))
+            else (attention_layer_activation,))
+      else:
+        attention_layer_activations = tuple(None for _ in attention_layer_sizes)
+      if len(attention_layer_activations) != len(attention_layer_sizes):
+        raise ValueError(
+            "If provided, attention_layer_activation must contain exactly one "
+            "activation function per attention layer, saw: %d vs %d"
+            % (len(attention_layer_activations), len(attention_layer_sizes)))
+      attention_layers_param = zip(
+          attention_layer_sizes, attention_layer_activations)
       self._attention_layers = tuple(
           layers_core.Dense(
-              attention_layer_size,
```

Comment:
I wonder if it makes sense to just allow a list of attention_layers as an alternative to passing attention_layer_size.  Then users could pass their own Dense (or something else) here.  what do you think?

Reply:
That makes sense. Do you still want to keep compatibility with the existing behavior? It seems cleaner and more powerful to only allow passing a list of `Layers`. I suppose you suggest to add a new option and raise an exception when both are set.